wolfcrypt/src/asn.c

```diff
@@ -3283,6 +3283,10 @@ static int GetIntPositive(mp_int* mpi, const byte* input, word32* inOutIdx,
     if (ret != 0)
         return ret;
 
+    if (idx < 1 || idx >= maxIdx) {
```

Comment:
Can't check for 1 as the starting index may not be zero.
GetASNInt ensures that a header (2 bytes) is read.
The length may be 0 though. and this is a problem.

The index won't be greater than maxIdx but the idx + length is not checked.

Reply:
Can you suggest a change please?

Reply:
Checked the code again and GetASNInt calls GetASNHeader and that will check that the length corresponds to bytes in the buffer. So no need to check idx + length > maxIdx.

if (length > 0) {
   /* Check that the preceding byte is zero when top bit set. */
   if (((input[idx] & 0x80) == 0x80) && (input[idx - 1] != 0x00))
         return MP_INIT_E;
}

Note that when the first byte of data has the top bit set, the preceding byte is the length and must be > 0.